Gems/Terrain/Code/Source/Components/TerrainSurfaceDataSystemComponent.cpp

```diff
@@ -240,6 +241,8 @@ namespace Terrain
             // Start listening for surface data events
             AZ_Assert((m_providerHandle != SurfaceData::InvalidSurfaceDataRegistryHandle), "Invalid surface data handle");
             SurfaceData::SurfaceDataProviderRequestBus::Handler::BusConnect(m_providerHandle);
+
+            SurfaceData::SurfaceDataTagProviderRequestBus::Handler::BusConnect();
```

Comment:
This should happen in Activate(), since the tags should be provided regardless of whether or not there's currently a terrain region.  There should also be a matching BusDisconnect() in Deactivate().

Reply:
That makes sense, changed.